store/displayPreferences.ts

```diff
@@ -0,0 +1,224 @@
+import { DisplayPreferencesDto } from '@jellyfin/client-axios';
+import { MutationTree, ActionTree, GetterTree } from 'vuex';
+import nuxtConfig from '~/nuxt.config';
+
+const stringToBoolean = (value: string) => value === 'True';
```

Comment:
Can you please explain what this function aims to solve?

Reply:
stringToBoolean? The `CustomPrefs` property of the `displayPreferences` as defined by the server is an object of keys to string value. The boolean values are also noted as `True` and `False` (although we could use whatever we want I guess) but in the end it's still an object with string values. This small function is just there to avoid repetition and be easier to edit the logic if we ever change the keyword for `true` and `false` bools.

Reply:
Ok, so the server returns boolean as a string? This c#.... always something has to be out of the standards :D

Reply:
The `displayPreferences` endpoint returns an object with a mix of different types, but the `CustomPrefs` object in it can only have string values. I think that's because in `CustomPrefs` no key is necessary, and maybe C# doesn't handle non specified objects with multiple types? I don't know in the end haha. Here is an exemple by the doc:

```json
{
  "Id": "string",
  "ViewType": "string",
  "SortBy": "string",
  "IndexBy": "string",
  "RememberIndexing": true,
  "PrimaryImageHeight": 0,
  "PrimaryImageWidth": 0,
  "CustomPrefs": {
    "additionalProp1": "string",
    "additionalProp2": "string",
    "additionalProp3": "string"
  },
  "ScrollDirection": "Horizontal",
  "ShowBackdrop": true,
  "RememberSorting": true,
  "SortOrder": "Ascending",
  "ShowSidebar": true,
  "Client": "string"
}
```

Reply:
I'll just resolve this conversation :)